### PR TITLE
[Serve][Doc] Setup Request Id

### PR DESCRIPTION
**Pull request by sihanwang41**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/serve/monitoring.md [hunk 1]
```diff
@@ -138,6 +138,23 @@ You can enable JSON-formatted logging in the Serve log file by setting the envir
 {"levelname": "INFO", "asctime": "2023-07-17 10:34:25,441", "deployment": "default_api", "replica": "default_api#bFDOnw", "request_id": "jLTczxOqme", "route": "/app1", "application": "default", "message": "replica.py:691 - __CALL__ OK 0.1ms"}
 ```
 
+### Set Request ID
+You can set a custom request ID by setting `RAY_SERVE_REQUEST_ID` in each request headers. For example
```
Comment (zcin) [inline doc/source/serve/monitoring.md:142]:
```suggestion
You can set a custom request ID for each HTTP request by including `RAY_SERVE_REQUEST_ID` in the request header. For example
```

File: doc/source/serve/monitoring.md [hunk 2]
```diff
@@ -138,6 +138,23 @@ You can enable JSON-formatted logging in the Serve log file by setting the envir
...
+```{literalinclude} doc_code/monitoring/request_id.py
+:language: python
+```
+Request ID `123-234` is printed in the http proxy log file and deployment log file, and the request ID is also included in the response.
```
Comment (zcin) [inline doc/source/serve/monitoring.md:146]:
```suggestion
The custom request ID `123-234` can be seen in the access logs that are printed to the HTTP Proxy log files and deployment log files.
```

File: doc/source/serve/monitoring.md [hunk 3]
```diff
@@ -138,6 +138,23 @@ You can enable JSON-formatted logging in the Serve log file by setting the envir
...
+HTTP proxy log file:
+```
+INFO 2023-07-20 13:47:54,221 http_proxy 127.0.0.1 123-234 / default http_proxy.py:538 - GET 200 8.9ms
+```
+
+Deployment log file:
+```
+(ServeReplica:default_Model pid=84006) INFO 2023-07-20 13:47:54,218 default_Model default_Model#yptKoo 123-234 / default replica.py:691 - __CALL__ OK 0.2ms
+```
+
```
Comment (zcin) [inline doc/source/serve/monitoring.md:157]:
```suggestion
The custom request ID is also included in the HTTP response.

[ example ]
```

**Review by zcin — COMMENTED** [review]
Overall looks good, some nits! Also, since you mentioned it, could you give an example of how to fetch the request id from an http response?

File: doc/source/serve/doc_code/monitoring/request_id.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+@serve.deployment
+class Model:
+    def __call__(self) -> int:
+        return 1
+
+
+serve.run(Model.bind())
+resp = requests.get(
+    "http://localhost:8000", headers={"RAY_SERVE_REQUEST_ID": "123-234"}
```
Comment (shrekris-anyscale) [inline doc/source/serve/doc_code/monitoring/request_id.py:14]: Are there any limitations on the request id's format (e.g. alphanumeric)? Is there any undefined behavior if two requests share the same id?

Comment (sihanwang41) [reply doc/source/serve/doc_code/monitoring/request_id.py:14]: No, we don't have limitations. Currently the request can be any chars string. When user setting the request id, relying on the customer to make it as unique, there is no undefined behavior, but you will see duplicate request id in the log.

**Comment by sihanwang41** [comment]
> Overall looks good, some nits! Also, since you mentioned it, could you give an example of how to fetch the request id from an http response?

It is inside the example :)

**Review by zcin — APPROVED** [review]

File: doc/source/serve/doc_code/monitoring/request_id.py (hunk 4 above)
Comment (shrekris-anyscale) [reply doc/source/serve/doc_code/monitoring/request_id.py:14]: Sounds good!

**Review by shrekris-anyscale — APPROVED** [review]
Nice work!

**Pull request merged**